src/posted_message.rs

```diff
@@ -0,0 +1,175 @@
+use {
+    borsh::{
+        BorshDeserialize,
+        BorshSerialize,
+    },
+    serde::{
+        Deserialize,
+        Serialize,
+    },
+    std::{
+        io::{
+            Error,
+            ErrorKind::InvalidData,
+            Write,
+        },
+        ops::{
+            Deref,
+            DerefMut,
+        },
+    },
+};
+
+#[repr(transparent)]
+#[derive(Default)]
+pub struct PostedMessageData {
+    pub message: MessageData,
+}
+
+#[derive(Default, Debug)]
+pub struct PostedMessageUnreliableData {
+    pub message: MessageData,
+}
```

Comment:
we are going to only process unreliable data. i recommend removing PostedMessageData and its implementation